### PR TITLE
flask: initial integration

### PR DESCRIPTION
**Pull request by DavidKorczynski**

### CODE REVIEW

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Comment by davidism** [comment]
Hi @jonathanmetzman, it looks like @DavidKorczynski might not actually be related to Google, so I want to make you aware that these automated tools come off as spam and don't feel good to get given the lack of direct commitment in the message. I'm copying what I said in https://github.com/pallets/jinja/issues/1318, https://github.com/pallets/werkzeug/issues/2011, and https://github.com/pallets/flask/issues/4538 (none of which received response) here. Please pass it up to someone on your team who can work with me directly if that's not you.

---

Thanks, this looks interesting, however I don't have the time to manage another service like this. I know Google uses our projects extensively internally and externally, and I'd be happy to accept PRs their employees find as part of this program, all contributions are welcome. :-) Thank you.

I am open to contributions from Google, but as a company they can surely support the most used packages in the Python ecosystem more directly and thoughtfully than this.

**Comment by jonathanmetzman** [comment]
@davidism Apologies for the spam. @DavidKorczynski is actually a contractor for Google. I guess you've gotten a lot of these messages because you are a prolific maintainer of many important libraries.
I can work with you directly if you have any questions.
We totally understand if you don't have time for these, at the end of the day we will respect your decision to include or not include your libraries in OSS-Fuzz.
We can support your tools if they use OSS-Fuzz, through [OSS-Fuzz integration rewards](https://google.github.io/oss-fuzz/getting-started/integration-rewards/) which are cash rewards in return for maintaining certain levels of fuzzing in your projects.
Please let us know if you'd like to proceed with this integration or discuss this further. Otherwise I can close this PR.
Sorry again.

**Comment by oliverchang** [comment]
Hi @davidism! Sorry for the spam here. As @jonathanmetzman mentioned, @DavidKorczynski is a contractor for us helping us find bugs in OSS via OSS-Fuzz integrations. Given the large volume they do, their outreach may feel a bit generic. Sorry about that. 

We can go ahead with this integration without bugging you, and if we do find any bugs, we'll send you actionable/reproducible bug reports :)

File: projects/flask/fuzz_werkzeug_http.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import atheris
+import sys
+
+with atheris.instrument_imports():
+  import werkzeug.http as whttp
+
+
+def TestOneInput(data):
+  fdp = atheris.FuzzedDataProvider(data)
+  whttp.parse_content_range_header(fdp.ConsumeUnicode(100))
```
Comment (oliverchang) [inline projects/flask/fuzz_werkzeug_http.py:25]: Any dictionaries we can use for http headers?

**Review by oliverchang — APPROVED** [review]

**Review by AdamKorcz — APPROVED** [review]

**Comment by davidism** [comment]
I'd be happy to have a call with you all some time to discuss this more. I'll be at PyCon if anyone wants to find me there.

Fuzzing Flask is probably not the best target. You'd probably get better results by fuzzing the other Pallets libraries it depends on: Werkzeug, Jinja, Click, MarkupSafe, and ItsDangerous. Flask itself does very little besides wrap various calls to those libraries.

**Pull request merged**